src/alpine/3.17/helix/amd64/Dockerfile

```diff
@@ -1,33 +1,51 @@
-FROM mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.17-local
+FROM alpine:3.17
+# Install .NET and test dependencies
 RUN apk update && apk add --no-cache \
-    cargo \
+    bash \
+    coreutils \
+    curl \
+    icu-data-full \
     iputils \
-    libffi-dev \
+    lttng-ust \
     musl-locales \
-    rust
+    openssl \
+    python3 \
+    sudo \
+    tzdata
 
 # Install Helix Dependencies
-RUN ln -sf /usr/bin/python3 /usr/bin/python && \
+RUN apk update && apk add --no-cache && \
+    apk add cargo linux-headers python3-dev openssl-dev && \
+    ln -sf /usr/bin/python3 /usr/bin/python && \
     curl https://bootstrap.pypa.io/get-pip.py -o ./get-pip.py --fail --silent --show-error && \
-    python ./get-pip.py && rm ./get-pip.py && \
-    python -m pip install --upgrade pip==22.2.2 && \
-    python -m pip install virtualenv==20.16.5 && \
+    python3 ./get-pip.py && rm ./get-pip.py && \
+    python3 -m pip install --upgrade pip==22.2.2 && \
+    python3 -m pip install virtualenv==20.16.5 && \
     pip download --no-deps helix-scripts --index-url https://dnceng.pkgs.visualstudio.com/public/_packaging/helix-client-prod/pypi/simple && \
-    pip install ./helix_scripts-*-py3-none-any.whl
+    pip install ./helix_scripts-*-py3-none-any.whl && \
+    apk del cargo linux-headers python3-dev openssl-dev
```

Comment:
Helix does a bunch of wheel management in it's start up scripts. I'm worried removing these packages might break something in some weird corner of some servicing branch.

Reply:
Right. This is one reason why I chose Alpine 3.17 - it is brand new and even runtime did not use it until very recently.

Reply:
I think the concern here is about helix itself, not the branches that are running. If *helix* needs these packages installed to do something in some niche way, we'll need them to stick around

Reply:
Is there some way how to determine that @michellemcdaniel? As I mentioned I'm planning to run tests but the questions if that would show up. AFAIK this is currently only used by runtime in `extra-platforms` pipeline that runs rarely and probably can take some risk. 

This does not block us in any way, so I can just leave it as is. I was trying to check if we can come up with smaller images - but it should OK if we cannot it. Alpine in particular is focus on small footprint so seeing 2G made me wonder.  

We _could_ also leave `rust` or whatever Helix needs there for now. We would still end up with much smaller size.

Reply:
The only way I can think of for confirming that is to test it. Like publish a version of it and then run it through a pipeline. What if instead of modifying this dockerfile, we created a scouting file/version with these changes so that we won't break anyone who might be using this and then we can test this version out in various pipelines so we can see what can and can't be removed? And then once we've done all the scouting, we can make changes to the main version of the file and remove the scouting version. I don't know how @mthalman would feel about that though.

Reply:
we could. AFAIK this is only consumed by https://github.com/dotnet/runtime/pull/81841 - the `extra-platform` only runs on demand and AFAIK it had 0% success rate in last few month. 
I was also thinking about leaving the `cargo` in but leave it in the Helix layer to make it clear it is Helix dependency.
At this point I'm trying to stabilize the pattern and duplicated for other architectures.

Reply:
BTW we could simply not promote it to production, right? e.g. leave it in staging until proven good.

Reply:
Not really. That would block every change that is made to this repo from rolling out. The best we can do is check in on a Tuesday after the rollout PR is made (assuming nothing else is necessary, so really, Wednesday is best), and then that would give you a week to test it and iterate (hopefully quickly) before the next rollout would kind of require it to be reverted, assuming that there is something else that actually needs to be rolled out.

Reply:
If this is just a one-time thing, and not going to be a regular occurrence, I can build this Dockerfile and push it to our dev ACR. You can then pull it from there. If this is going to happen more often, my suggestion would be for the runtime team to have its own ACR for these testing purposes.

Reply:
> duplicated for other architectures

Are you planning on making more sweeping changes? Remember, most of this repository is going to be discarded in a month or two. _Please_ coordinate with @michellemcdaniel and I, otherwise all of this work is going to get discarded along with it.  Now is not a good time for large changes to this repository.  Since we are in the middle of attempting to migrate it, it's very likely any changes will be lost.

Reply:
I don't see this as sweeping change @ChadNedzlek. This is minor cleanup to avoid pulling in unneeded packages. And no, I don't plan more changes like this. We will need to add MsQuic build to more images but that does not really change the infrastructure.